app/flow_io/flow/order.rb

```diff
@@ -150,6 +150,14 @@ def build_flow_request
     body[:selection] = [@order.flow_data['selection']]         if @order.flow_data['selection']
     body[:delivered_duty] = @order.flow_data['delivered_duty'] if @order.flow_data['delivered_duty']
 
+    # discount on full order is applied
+    if @order.adjustment_total != 0
+      body[:discount] = {
+        amount: @order.adjustment_total,
+        currency: @order.currency
```

Comment:
yes, `@order` is instance of `Spree::Order` so that is local Solidus base currency, USD. I checked.